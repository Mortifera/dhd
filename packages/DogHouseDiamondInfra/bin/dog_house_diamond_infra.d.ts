#!/usr/bin/env node
import 'source-map-support/register';
import * as cdk from 'monocdk';
export declare const app: cdk.App;
