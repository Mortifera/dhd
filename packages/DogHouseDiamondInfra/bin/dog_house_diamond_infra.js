#!/usr/bin/env node
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.app = void 0;
require("source-map-support/register");
const cdk = require("monocdk");
const StreamDataSyncStack_1 = require("../lib/StreamDataSyncStack");
exports.app = new cdk.App();
new StreamDataSyncStack_1.StreamDataSyncStack(exports.app, 'DHDStreamDataSyncStack');
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiZG9nX2hvdXNlX2RpYW1vbmRfaW5mcmEuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJkb2dfaG91c2VfZGlhbW9uZF9pbmZyYS50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7O0FBQ0EsdUNBQXFDO0FBQ3JDLCtCQUErQjtBQUMvQixvRUFBaUU7QUFFcEQsUUFBQSxHQUFHLEdBQUcsSUFBSSxHQUFHLENBQUMsR0FBRyxFQUFFLENBQUM7QUFDakMsSUFBSSx5Q0FBbUIsQ0FBQyxXQUFHLEVBQUUsd0JBQXdCLENBQUMsQ0FBQyIsInNvdXJjZXNDb250ZW50IjpbIiMhL3Vzci9iaW4vZW52IG5vZGVcbmltcG9ydCAnc291cmNlLW1hcC1zdXBwb3J0L3JlZ2lzdGVyJztcbmltcG9ydCAqIGFzIGNkayBmcm9tICdtb25vY2RrJztcbmltcG9ydCB7IFN0cmVhbURhdGFTeW5jU3RhY2sgfSBmcm9tICcuLi9saWIvU3RyZWFtRGF0YVN5bmNTdGFjayc7XG5cbmV4cG9ydCBjb25zdCBhcHAgPSBuZXcgY2RrLkFwcCgpO1xubmV3IFN0cmVhbURhdGFTeW5jU3RhY2soYXBwLCAnREhEU3RyZWFtRGF0YVN5bmNTdGFjaycpO1xuIl19