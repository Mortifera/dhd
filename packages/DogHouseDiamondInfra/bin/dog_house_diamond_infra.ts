#!/usr/bin/env node
import 'source-map-support/register';
import * as cdk from 'monocdk';
import { DHDStack } from "../lib/DHDStack";

export const app = new cdk.App();
new DHDStack(app, 'DogHouseDiamondStack');
