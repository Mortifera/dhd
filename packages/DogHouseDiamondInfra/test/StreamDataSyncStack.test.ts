import * as DogHouseDiamondInfra from '../bin/dog_house_diamond_infra';

const synthedApp = DogHouseDiamondInfra.app.synth();

test('Stacks', () => {
    expect(synthedApp.stacks.map(x => x.stackName)).toMatchSnapshot();
});

synthedApp.stacks.forEach(stack => {
    test('Stack - ' + stack.stackName, () => {
        expect(stack.template).toMatchSnapshot();
    });
});