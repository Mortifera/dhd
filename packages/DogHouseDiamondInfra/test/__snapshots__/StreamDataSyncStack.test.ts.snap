// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack - DogHouseDiamondStack 1`] = `
Object {
  "Outputs": Object {
    "BucketArn": Object {
      "Value": Object {
        "Fn::GetAtt": Array [
          "DataSyncCE22D76D",
          "Arn",
        ],
      },
    },
  },
  "Resources": Object {
    "DataSyncCE22D76D": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "DataSyncUser213EE724": Object {
      "Type": "AWS::IAM::User",
    },
    "DataSyncUserDefaultPolicyDA4A5DB9": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "DataSyncCE22D76D",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "DataSyncCE22D76D",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "DataSyncCE22D76D",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PassTable33D248DE",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PassTableKey40C4EE0A",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "dynamodb:BatchWriteItem",
                "dynamodb:PutItem",
                "dynamodb:UpdateItem",
                "dynamodb:DeleteItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "PassTable33D248DE",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
            Object {
              "Action": Array [
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PassTableKey40C4EE0A",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PassTableKey40C4EE0A",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "DataSyncUserDefaultPolicyDA4A5DB9",
        "Users": Array [
          Object {
            "Ref": "DataSyncUser213EE724",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PassTable33D248DE": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "service",
            "AttributeType": "S",
          },
        ],
        "KeySchema": Array [
          Object {
            "AttributeName": "service",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": Object {
          "ReadCapacityUnits": 5,
          "WriteCapacityUnits": 5,
        },
        "SSESpecification": Object {
          "KMSMasterKeyId": Object {
            "Fn::GetAtt": Array [
              "PassTableKey40C4EE0A",
              "Arn",
            ],
          },
          "SSEEnabled": true,
          "SSEType": "KMS",
        },
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "PassTableKey40C4EE0A": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "Description": "Customer-managed key auto-created for encrypting DynamoDB table at DogHouseDiamondStack/PassTable",
        "EnableKeyRotation": true,
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion",
                "kms:GenerateDataKey",
                "kms:TagResource",
                "kms:UntagResource",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "DataSyncUser213EE724",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "DataSyncUser213EE724",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
            Object {
              "Action": Array [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::GetAtt": Array [
                    "DataSyncUser213EE724",
                    "Arn",
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;

exports[`Stacks 1`] = `
Array [
  "DogHouseDiamondStack",
]
`;
